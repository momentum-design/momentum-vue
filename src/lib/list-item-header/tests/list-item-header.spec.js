import { mount } from '@vue/test-utils';
import ListItemHeader from '../index.vue';
import ListItem from '../../list-item/index.vue';
import ListItemSection from '../../list-item-section/index.vue';
import Vue from 'vue';

Vue.component(ListItem.name, ListItem);
Vue.component(ListItemSection.name, ListItemSection);

describe('ListItemHeader', () => {
  it('should match snapshot', () => {
    const wrapper = mount(ListItemHeader, {
      propsData: {
        header: 'header'
      }
    });
    expect(wrapper.html()).toMatchSnapshot();
  });

  it('should render children', () => {
    const wrapper = mount(ListItemHeader, {
      propsData: {
        header: 'header'
      },
      slots: {
        default: '<div class="test">Test</div>'
      }
    });

    expect(wrapper.findAll('.test').length).toEqual(1);
  });

  it('should render one ListItemHeader', () => {
    const wrapper = mount(ListItemHeader, {
      propsData: {
        header: 'header'
      }
    });

    expect(wrapper.classes('md-list-item-header')).toEqual(true);
  });

  it('should handle isReadOnly prop', () => {
    const wrapper = mount(ListItemHeader, {
      propsData: {
        header: 'header',
        isReadOnly: true
      }
    });

    expect(wrapper.props().isReadOnly).toEqual(true);
  });

  it('should allow children to be clicked on', () => {
    let count = 0;
    const clickableNode = '<div class="testClick" role="button" @click="onClickFn" @keydown="onClickFn" tabindex="0"/>';
    const wrapper = mount(ListItemHeader, {
      propsData: {
        header: 'Testing'
      },
      slots: {
        default: clickableNode
      },
      mocks: {
        onClickFn: () => { count++; }
      }
    });

    wrapper.find('.testClick').trigger('click');
    expect(count).toEqual(1);
  });

  it('should handle type prop', () => {
    const wrapper = mount(ListItemHeader, {
      propsData: {
        type: 'space',
        header: 'header'
      }
    });

    expect(wrapper.find('.md-list-item-header--space').exists()).toEqual(true);
  });

  it('should handle className prop', () => {
    const wrapper = mount(ListItemHeader, {
      propsData: {
        class: 'menuItem',
        header: 'header'
      }
    });

    expect(wrapper.classes('menuItem')).toEqual(true);
  });

  it('should handle header prop', () => {
    const wrapper = mount(ListItemHeader, {
      propsData: {
        header: 'header'
      }
    });

    expect(wrapper.find('.md-list-item__header').text()).toEqual('header');
  });

  it('should pass props to ListItem', () => {
    const wrapper = mount(ListItemHeader, {
      propsData: {
        header: 'header',
        separator: true
      }
    });

    expect(wrapper.classes('md-list-item--separator')).toEqual(true);
  });

  describe('tests for title Prop', () => {
    it('should handle title prop', () => {
      const wrapper = mount(ListItemHeader, {
        propsData: {
          header: 'header',
          title: 'testTitle'
        }
      });

      expect(wrapper.attributes().title).toEqual('testTitle');
    });

    it('should handle title if header is string', () => {
      const wrapper = mount(ListItemHeader, {
        propsData: {
          header: 'testTitle'
        }
      });

      expect(wrapper.attributes().title).toEqual('testTitle');
    });
  });
});

