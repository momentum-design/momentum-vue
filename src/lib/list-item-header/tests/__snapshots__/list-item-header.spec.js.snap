// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListItemHeader should match snapshot 1`] = `
<div data-md-event-key="md-space-list-item__header-1" role="listitem" ref="navLink" data-md-keyboard-key="header" title="header" eventkey="" class="md-list-item md-list-item--read-only md-list-item-header" id="md-space-list-item__header-1">
  <div class="md-list-item__center">
    <div class="md-list-item__header">
      header
    </div>
  </div>
  <div class="md-list-item__right"></div>
</div>
`;
